### PR TITLE
Update miscfiles_manage_all_certs() to include creating directories

### PR DESCRIPTION
**Pull request by zpytela**
Update the miscfiles_manage_all_certs() so that it contains also rules
for creating directories.

### CODE REVIEW

**Comment by WOnder93** [comment]
So I ran the reproducer from the slapd BZ in permissive mode and got these AVCs:
```
----
type=PROCTITLE msg=audit(30.10.2020 05:02:09.839:740) : proctitle=/usr/sbin/slapd -u ldap -h ldapi:/// ldap:/// ldaps:/// 
type=SYSCALL msg=audit(30.10.2020 05:02:09.839:740) : arch=x86_64 syscall=mkdir success=yes exit=0 a0=0x563546756300 a1=0700 a2=0x6 a3=0x5f9bd691 items=0 ppid=1 pid=24480 auid=unset uid=ldap gid=ldap euid=ldap suid=ldap fsuid=ldap egid=ldap sgid=ldap fsgid=ldap tty=(none) ses=unset comm=slapd exe=/usr/sbin/slapd subj=system_u:system_r:slapd_t:s0 key=(null) 
type=AVC msg=audit(30.10.2020 05:02:09.839:740) : avc:  denied  { create } for  pid=24480 comm=slapd name=dbTemp.frAwHS scontext=system_u:system_r:slapd_t:s0 tcontext=system_u:object_r:slapd_cert_t:s0 tclass=dir permissive=1 
type=AVC msg=audit(30.10.2020 05:02:09.839:740) : avc:  denied  { add_name } for  pid=24480 comm=slapd name=dbTemp.frAwHS scontext=system_u:system_r:slapd_t:s0 tcontext=system_u:object_r:slapd_cert_t:s0 tclass=dir permissive=1 
type=AVC msg=audit(30.10.2020 05:02:09.839:740) : avc:  denied  { write } for  pid=24480 comm=slapd name=certs dev="vda1" ino=522232 scontext=system_u:system_r:slapd_t:s0 tcontext=system_u:object_r:slapd_cert_t:s0 tclass=dir permissive=1 
----
type=PROCTITLE msg=audit(30.10.2020 05:02:09.861:741) : proctitle=/usr/sbin/slapd -u ldap -h ldapi:/// ldap:/// ldaps:/// 
type=SYSCALL msg=audit(30.10.2020 05:02:09.861:741) : arch=x86_64 syscall=rmdir success=yes exit=0 a0=0x563546756300 a1=0x7fffc92992d0 a2=0x3cc a3=0x4 items=0 ppid=1 pid=24480 auid=unset uid=ldap gid=ldap euid=ldap suid=ldap fsuid=ldap egid=ldap sgid=ldap fsgid=ldap tty=(none) ses=unset comm=slapd exe=/usr/sbin/slapd subj=system_u:system_r:slapd_t:s0 key=(null) 
type=AVC msg=audit(30.10.2020 05:02:09.861:741) : avc:  denied  { rmdir } for  pid=24480 comm=slapd name=dbTemp.frAwHS dev="vda1" ino=8388710 scontext=system_u:system_r:slapd_t:s0 tcontext=system_u:object_r:slapd_cert_t:s0 tclass=dir permissive=1 
type=AVC msg=audit(30.10.2020 05:02:09.861:741) : avc:  denied  { remove_name } for  pid=24480 comm=slapd name=dbTemp.frAwHS dev="vda1" ino=8388710 scontext=system_u:system_r:slapd_t:s0 tcontext=system_u:object_r:slapd_cert_t:s0 tclass=dir permissive=1 
```
So both this and the slapd fix will need also permissions for removing the temporary subdirectory, not just creating it.

File: policy/modules/system/miscfiles.if [hunk 1]
```diff
@@ -82,6 +82,7 @@ interface(`miscfiles_manage_all_certs',`
 	')
 
 	allow $1 cert_type:dir list_dir_perms;
```
Comment (WOnder93) [inline policy/modules/system/miscfiles.if]: This is not strictly relevant, but while you're touching this code, you could also remove this line, which is a subset of the interface calls below (it was so already before this commit).

**Comment by zpytela** [comment]
The miscfiles_manage_all_certs() now contains manage rules which contain all permissions:
```

define(`manage_dirs_pattern',`
        allow $1 $2:dir rw_dir_perms;
        allow $1 $3:dir manage_dir_perms;
')
define(`manage_dir_perms',`{ create open getattr setattr read write link unlink rename search add_name remove_name reparent rmdir lock ioctl }')
```

**Comment by WOnder93** [comment]
One more comment: please mention in the pull request that this is needed because of a change in NSS.

**Review by WOnder93 — APPROVED** [review]
LGTM, thanks!

**Comment by zpytela** [comment]
Thank you for the deep-dive review.

**Pull request merged**